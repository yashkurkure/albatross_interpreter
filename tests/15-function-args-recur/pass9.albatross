var result int := 0;

fun factorial int (i int) {
  if (i == 0) { return 1; }
  else {
    result := i * factorial(i-1);
    return result;
  }
}

repeat(10) {
    factorial(10);
    factorial(5);
    printint(result);
    printstring(" ");
    factorial(10);
    printint(result);
    printstring(" ");
}
return 0;
