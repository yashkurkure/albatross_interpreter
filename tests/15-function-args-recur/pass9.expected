120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 120 3628800 
